torch/distributed/_tensor/device_mesh.py

```diff
@@ -121,32 +123,37 @@ def __init__(
             self._dim_groups = self._init_process_groups()
 
     def _get_or_create_default_group(self):
-        self._backend = "gloo" if self.device_type == "cpu" else "nccl"
-        if not is_initialized():
+        self._backend = Backend.GLOO if self.device_type == "cpu" else Backend.NCCL
+        default_initialized = is_initialized()
+        if not default_initialized:
             init_process_group(backend=self._backend)
-        else:
-            world_size = get_world_size()
-            if self.mesh.numel() > world_size:
-                raise RuntimeError(
-                    f"Mesh should not be bigger than default world size, but found {self.mesh.numel()} ranks!"
-                )
+
+        world_size = get_world_size()
+        if self.mesh.numel() > world_size:
+            raise RuntimeError(
+                f"Mesh should not be bigger than default world size, but found {self.mesh.numel()} ranks!"
+            )
```

Comment:
Thx for the fix. Because sub-mesh is not contiguous, comparing `numel()` with world_size is not enough but `max()` should do the thing. We can change `numel()` to `max()`.

Reply:
We are only creating default pg here but not submesh, so we shouldn't check anything related with submesh